### ISSUE TITLE
Can't get grav to run on my server

### ISSUE THREAD

**Issue by Yandols** [getgrav/grav#2205]
Hi, I am trying to get grav on my site because I have been looking high and low for a flatfile cms system for ages and finally found out about grav, but I am currently having problems getting it to run. 

I have tried all 3 methods of installing grav onto my server, all of them succeeded properly but whenever I tried to access it with my browser I just get a 500 Internal Server Error. 

I looked up the troubleshooting page but nothing there is able to solve my problem so I have decided to open an issue about it and hope I can get some insight on what I am doing wrong here.

Some Information about my server are as follows:
```
OS: Fedora 27 Cloud Edition
Kernel: Linux 4.15.15-300.fc27.x86_64
Running: Apache 2.4 on httpd
PHP 7.1.16 (cli) (built: Mar 28 2018 07:11:55) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2018 Zend Technologies
```

Hopefully this is the right place for me to ask about this problem. I really want to switch over to grav from jekyll because it's hard for me to keep getting jekyll to push out page updates in my use case.

Thanks in advance,
Yandols.

**Comment by Perlkonig** [comment]
A 500 should produce a server log entry. What does the log say?

**Comment by Yandols** [comment]
The strange thing is that there's nothing in my server error log that relates to grav. The last error in the log is for something else completely unrelated to grav.

At first I thought it could be a problem with a missing PHP dependency as well, but when I tried installing them via `dnf` I already have all of them installed and up-to-date.

The problems plugin is indeed under `/grav/user/plugins/problems` so that does exist. I'm not sure what the problem is.

**Comment by Perlkonig** [comment]
And if it were a missing dependency, the log should say as much. If Apache throws a 500, it will have *something* to say. The 500 *could* be unrelated to Grav. What you do is `tail` the log and note the timestamp of the last entry. Refresh your Grav page and trigger the 500. Then `tail` the log again. There should be a new entry. Whether it seems related to Grav or not, that's where you want to start. There are lots of things that could be a problem, and we could list them (`.htaccess` file, for example), but shooting in the dark is not an efficient way to proceed. Let's find out what the server is complaining about.

**Comment by Yandols** [comment]
That's the weird part. I accessed the page again and it returned the 500 error, checked my server log and the last error is still the same one from another application of mine which occured a day before I installed Grav. Grav didn't generate anything into the server log. I don't think there's anything wrong with my permissions either since it's already set to 0775.

**Comment by Perlkonig** [comment]
This is indeed weird. I'm not aware of a scenario where no log entry is created. Are you seeing anything in `[GRAVROOT]/user/logs/grav.log`?

**Comment by Yandols** [comment]
Unfortunately that file does not exist in my grav installation directory. Not even the `logs` directory under `user` for that matter.

**Comment by Yandols** [comment]
The only file in that folder is an empty `.gitkeep` file. Unfortunately there are no logs inside of that folder either.

I've been scratching my head on how to even find out what's wrong with this, so I decided to put my apache log level to `debug` logging instead of `warn` which is what I normally have it set to. 

```
[Thu Oct 04 03:41:28.601890 2018] [proxy:debug] [pid 27147:tid 139640384218880] mod_proxy.c(1234): [client [ip-address-redacted]:63670] AH01143: Running scheme unix handler (attempt 0)
[Thu Oct 04 03:41:28.601898 2018] [proxy_ajp:debug] [pid 27147:tid 139640384218880] mod_proxy_ajp.c(742): [client [ip-address-redacted]:63670] AH00894: declining URL fcgi://localhost/var/www/html/yandols/gravitywell/index.php
[Thu Oct 04 03:41:28.601921 2018] [proxy_fcgi:debug] [pid 27147:tid 139640384218880] mod_proxy_fcgi.c(995): [client [ip-address-redacted]:63670] AH01076: url: fcgi://localhost/var/www/html/yandols/gravitywell/index.php proxyname: (null) proxyport: 0
[Thu Oct 04 03:41:28.601926 2018] [proxy_fcgi:debug] [pid 27147:tid 139640384218880] mod_proxy_fcgi.c(1002): [client [ip-address-redacted]:63670] AH01078: serving URL fcgi://localhost/var/www/html/yandols/gravitywell/index.php
[Thu Oct 04 03:41:28.601932 2018] [proxy:debug] [pid 27147:tid 139640384218880] proxy_util.c(2156): AH00942: FCGI: has acquired connection for (*)
[Thu Oct 04 03:41:28.601939 2018] [proxy:debug] [pid 27147:tid 139640384218880] proxy_util.c(2209): [client [ip-address-redacted]:63670] AH00944: connecting fcgi://localhost/var/www/html/yandols/gravitywell/index.php to localhost:8000
[Thu Oct 04 03:41:28.601944 2018] [proxy:debug] [pid 27147:tid 139640384218880] proxy_util.c(2246): [client [ip-address-redacted]:63670] AH02545: fcgi: has determined UDS as /run/php-fpm/www.sock
[Thu Oct 04 03:41:28.602038 2018] [proxy:debug] [pid 27147:tid 139640384218880] proxy_util.c(2418): [client [ip-address-redacted]:63670] AH00947: connected /var/www/html/yandols/gravitywell/index.php to httpd-UDS:0
[Thu Oct 04 03:41:28.602068 2018] [proxy:debug] [pid 27147:tid 139640384218880] proxy_util.c(2789): AH02823: FCGI: connection established with Unix domain socket /run/php-fpm/www.sock (*)
[Thu Oct 04 03:41:28.687322 2018] [proxy:debug] [pid 27147:tid 139640384218880] proxy_util.c(2171): AH00943: FCGI: has released connection for (*)
```

Unfortunately these are all just at debug level and not errors or warnings so it doesn't look very helpful, at least to me. But I've pasted them here in case you might find something from it. Where `/gravitywell/` is Grav's installation directory.

**Comment by Yandols** [comment]
Whoops, forgot to keep this issue updated.

A few weeks ago I did a system maintenance on my server and updated all my packages, including php. Some time later in the same week I tried accessing my grav installation again and to my surprise it greeted me with the owner account creation screen. 

So I think there was a problem with my apache installation, causing grav to not function for some reason. But since everything was automated I don't think I can provide further information on why it did not work.

In any case, it's working now and I've deployed multiple instances of grav across my domain. It's really nice to use compared to WP and anything else I've ever used.
